x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/utils.ts

```diff
@@ -115,3 +117,17 @@ export const splitResultFields = (resultFields: FieldResultSettingObject, schema
 
   return { textResultFields, nonTextResultFields };
 };
+
+export const areFieldsEmpty = (fields: FieldResultSettingObject) => {
+  const anyNonEmptyField = Object.values(fields).find((resultSettings) => {
+    return !isEmpty(resultSettings);
+  });
+  return !anyNonEmptyField;
+};
```

Comment:
You could probably re-write this to use a `for...in` loop, which would allow you to short-circuit immediately if you founnd a non-empty field, but we're not talking about 1,000,000 fields here.

Reply:
True. `.find` will return the first match it finds in an array so I believe it is equally efficient, fwiw.